include/swift/Sema/ConstraintSystem.h

```diff
@@ -2676,25 +2666,19 @@ class ConstraintSystem {
   /// we're exploring.
   SolverState *solverState = nullptr;
 
-  struct ArgumentInfo {
-    ArrayRef<Identifier> Labels;
-    Optional<unsigned> UnlabeledTrailingClosureIndex;
-  };
-
   /// A mapping from the constraint locators for references to various
   /// names (e.g., member references, normal name references, possible
-  /// constructions) to the argument labels provided in the call to
-  /// that locator.
-  llvm::DenseMap<ConstraintLocator *, ArgumentInfo> ArgumentInfos;
+  /// constructions) to the argument lists for the call to that locator.
+  llvm::DenseMap<ConstraintLocator *, ArgumentList *> ArgumentLists;
```

Comment:
While we are at it I think we need to do change this to behave like `addNodeTypes` and record `ArgumentLists` in the solution as well. Since constraint generation for result builders happens during solving this list would effectively get rewritten multiple times with the out-of-scope `ArgumentList` objects just to keep them around until constraint system lifetime ends.